src/main/java/UI/ListingDetailPage.java

```diff
@@ -1,20 +1,166 @@
 package UI;
 
+import Main.Main;
+import database.DatabaseController;
 import entities.Listing;
+import entities.User;
 import forms.AddToCartForm;
+import useCase.addToCart.AddToCartResponseModel;
+import database.DatabaseController.*;
 
-public class ListingDetailPage extends Page {
-    private Listing listing;
-    private boolean canAddToCart = true;
-    private AddToCartForm addToCartForm;
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
 
-    public ListingDetailPage(String title, Listing listing) {
-        super(title);
+public class ListingDetailPage extends Page implements ActionListener {
+    final int WIDTH = 1280;
+    final int HEIGHT = 570;
+
+    private final DatabaseController db = new DatabaseController<>();
+    private final Listing listing;
+
+    private JLabel message;
+    private JButton addToCartButton;
+
+    public ListingDetailPage(Listing listing) {
+        super(listing.getTitle());
         this.listing = listing;
-        // TODO: figure out currentUser
-        // this.addToCartForm = new AddToCartForm(currentUser.username, listing);
+        this.setLayout(null);
+
+        boolean canAddToCart = canAddToCart();
+        this.setPreferredSize(new Dimension(1280, 720));
+
+        User listingSeller = getListingSeller();
+        if (listingSeller == null) {
+            message.setLocation(500, 400);
+            message.setVisible(true);
+            this.add(message);
+            this.setVisible(true);
+            return;
+        }
+
+
+        JLabel listingTitle = new JLabel(listing.getTitle());
+        listingTitle.setSize(500, 60);
+        listingTitle.setLocation(500, 120);
+        listingTitle.setFont(new Font("Arial", Font.PLAIN, 50));
+        this.add(listingTitle);
+
+        try {
+            BufferedImage rawImage = ImageIO.read(new File("src/main/java/image.JPG"));
+            JLabel img = new JLabel(new ImageIcon(rawImage));
```

Comment:
Does this line throw an IOException to be caught? When I was making the logo button, I tried to read an image from a file path that didn't exist, and it didn't throw an error, it just displayed an empty image. Will need to test what displays when rawImage doesn't exist.

Reply:
i don't know. we will test this when the db methods all work so it's easier to write tests